Add RISC-V V Support

mp-17 opened the pull request.

zarubaf commented:
Wow that’s great! Yeah I think it would be nice to have our patches on top - so 👍 for rebasing (if it isn’t too much work)

niwis commented:
`occamy` is rebased, so this should be good to merge :-)

zarubaf commented:
Awesome! Thanks!

moimfeld committed:
Fix march string when compiling with a newer compiler

The issue was that when compiling u-boot from the cva6-sdk for Cheshire + Ara we used a newer compiler (GCC 13.2.0). This compiler complies with a newer version of the RISC-V specifications, where some instruction where removed from the base integer instruction set and moved into dedicated extensions (zicsr and zifence). In the cva6-sdk, when we compile for Cheshire + Ara, the RVV variable has to be defined, for this reason we check here if this variable is defined an change the march accordingly

Signed-off-by: Moritz Imfeld <moimfeld@student.ethz.ch>

moimfeld committed:
TODO: Check if this change is necessary

Signed-off-by: Moritz Imfeld <moimfeld@student.ethz.ch>

moimfeld committed:
Switch from ifdef to ifeq 1

Signed-off-by: Moritz Imfeld <moimfeld@student.ethz.ch>

mp-17 commented:
Already fixed on `cheshire` branch
